Add "ob-table-enhancer" plugin

Stardusten opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/Stardusten/ob-table-enhancer/tree/master

## Release Checklist
- [ ] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
[MyPlugin](https://github.com/Stardusten/ob-table-enhancer/blob/3ee82f5381efdf6f6436b46b59cb07b63f9ceb26/main.ts#L23)
Please rename this class to something more meaningful.

[// 劫持滚动事件](https://github.com/Stardusten/ob-table-enhancer/blob/3ee82f5381efdf6f6436b46b59cb07b63f9ceb26/main.ts#L55-L67)
This will only work for the editor instance that was loaded at start and will completely block scrolling.

[export const insertBelowIcon](https://github.com/Stardusten/ob-table-enhancer/blob/3ee82f5381efdf6f6436b46b59cb07b63f9ceb26/src/tool-bar.ts#L7)
Obsidian includes all icons from <https://lucide.dev/>, you can use any of those with the `setIcon` function from the API out of the box.

In general, you do way to much DOM manipulation,
most things you do here can either be acomplished with the Obsidian API or in a CodeMirror editor extension.
<https://marcus.se.net/obsidian-plugin-docs/editor/extensions>

Stardusten commented:
> [MyPlugin](https://github.com/Stardusten/ob-table-enhancer/blob/3ee82f5381efdf6f6436b46b59cb07b63f9ceb26/main.ts#L23)
Please rename this class to something more meaningful.

Fixed!

> [// 劫持滚动事件](https://github.com/Stardusten/ob-table-enhancer/blob/3ee82f5381efdf6f6436b46b59cb07b63f9ceb26/main.ts#L55-L67)
This will only work for the editor instance that was loaded at start and will completely block scrolling.

My bad, I've deleted it. But I have tried to prevent abnormal scrolling by using `dispatch` in codemirror editor instead (see `editor-utils.ts`), but but it still sometimes goes wrong. Do you have any good idea about this?

> [export const insertBelowIcon](https://github.com/Stardusten/ob-table-enhancer/blob/3ee82f5381efdf6f6436b46b59cb07b63f9ceb26/src/tool-bar.ts#L7)
Obsidian includes all icons from https://lucide.dev/, you can use any of those with the setIcon function from the API out of the box.

There are a few icons I need that I can't find on this site, so I'd rather not change them.

> In general, you do way to much DOM manipulation,
most things you do here can either be acomplished with the Obsidian API or in a CodeMirror editor extension.
https://marcus.se.net/obsidian-plugin-docs/editor/extensions

Indeed, I didn't  reading this document carefully before, but looked directly at the ob api, so I ignored the `EditorExtension` which is really useful and much more elegant than DOM manipulation. But this is difficult to modify now, I will try this when adding new features or refactoring later.

joethei commented:
Hi, we currently don't feel comfortable merging this in its current state.

The DOM hacks you do are to unstable to have a stable plugin in the long term.

From playing with it in my test vault on 0.16.2(current insider) I also found some issues with data synchronisation.
Basically the version visible in edit mode with your plugin and the actual version in the file did not match at all.

Sorry about that.

ericaxu commented:
Hi there, to keep things tidy, we're going to temporarily close plugin PRs after a month of inactivity.

Feel free to tell us to re-open it when you're ready to continue. Thanks for your understanding!

jwierzbo commented:
@Stardusten do you have any plans for code refactor to address that DOM manipulation change? Would be nice to have that plugin in the official plugin repo..
